packages/core-data/src/types/sidebar.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import { Widget } from './widget';
+import { Context } from './common';
+
+export interface Sidebar< C extends Context > {
+	/**
+	 * ID of sidebar.
+	 */
+	id: string;
+	/**
+	 * Unique name identifying the sidebar.
+	 */
+	name: string;
+	/**
+	 * Description of sidebar.
+	 */
+	description: string;
+	/**
+	 * Extra CSS class to assign to the sidebar in the Widgets interface.
+	 */
+	class: string;
+	/**
+	 * HTML content to prepend to each widget's HTML output when assigned to this sidebar. Default is an opening list item element.
+	 */
+	before_widget: string;
+	/**
+	 * HTML content to append to each widget's HTML output when assigned to this sidebar. Default is a closing list item element.
+	 */
+	after_widget: string;
+	/**
+	 * HTML content to prepend to the sidebar title when displayed. Default is an opening h2 element.
+	 */
+	before_title: string;
+	/**
+	 * HTML content to append to the sidebar title when displayed. Default is a closing h2 element.
+	 */
+	after_title: string;
+	/**
+	 * Status of sidebar.
+	 */
+	status: SidebarStatus;
+	/**
+	 * Nested widgets.
+	 */
+	widgets: NestedWidget< C >[];
```

Comment:
what about these is nested and what forms the nesting? it looks like it's a flat list and there's nothing inside of `Widget` to recurse and show nested values of the `Widget` type within it.

it may even be clearer to inline this type here given that we're showing it's a list of items which may be a widget or could be a string, supposedly a widget slug.

```ts
/**
 * Widgets (or their slugs) which are contained in this sidebar.
 */
widgets: (Widget<C> | string)[];
```

that way we don't have to redeclare the context parameter and it's clear by inspection that it could be a mix of widgets and strings. if I'm wrong about the slug part then obviously we should describe whatever is actually there. we can see this without scanning for another type.

this is a counter-balance to the `CommentStatus` type and a stylistic choice, so we have room to choose. in this particular case I found it clearer without the indirection.